tcms/rpc/tests/test_auth.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=attribute-defined-outside-init
+
+from xmlrpc.client import Fault as XmlRPCFault
+from django.contrib.sessions.models import Session
+from tcms.rpc.api.user import User
+from tcms.rpc.tests.utils import APITestCase
+from tcms.tests import UserFactory
+
+
+class TestAuthLogin(APITestCase):
+    """Test Auth.login method"""
+
+    def _fixture_setup(self):
+        super()._fixture_setup()
+        #
+        self.rpc_client.Auth.logout()
+        self.tester = UserFactory()
+        self.password = "password"
+        self.tester.set_password(self.password)
+        self.tester.save()
+
+    def test_login_with_username(self):
+        self.rpc_client.Auth.logout()
+        sesson_id = self.rpc_client.Auth.login(self.tester.username, self.password)
+        session = Session.objects.get(session_key=sesson_id).get_decoded()
+
+        self.assertIsNotNone(sesson_id)
+        self.assertEqual(User.objects.get(pk=session.get("_auth_user_id")), self.tester)
+
+    def test_login_without_username(self):
+        self.rpc_client.Auth.logout()
+        with self.assertRaisesRegex(XmlRPCFault, "Username and password is required"):
+            self.rpc_client.Auth.login("", self.password)
+
+    def test_login_without_password(self):
+        self.rpc_client.Auth.logout()
+        with self.assertRaisesRegex(XmlRPCFault, "Username and password is required"):
+            self.rpc_client.Auth.login(self.tester.username, "")
+
+    def test_login_with_incorrect_username(self):
+        self.rpc_client.Auth.logout()
+        with self.assertRaisesRegex(XmlRPCFault, "Wrong username or password"):
+            self.rpc_client.Auth.login("kiwi_tester", self.password)
+
+    def test_login_with_incorrect_password(self):
+        self.rpc_client.Auth.logout()
+        with self.assertRaisesRegex(XmlRPCFault, "Wrong username or password"):
+            self.rpc_client.Auth.login(self.tester.username, "kiwi-password")
```

Comment:
IMO only 1 of these scenarios is enough (or even login with empty string as the 2 values) b/c the work is done by django.contrib.auth.authenticate and running through all of the possible invalid combinations is a job for Django's test suite. 

Here we are interested in the fact that the result will be None and the API method will raise the expected exception.